server/src/test/java/io/crate/planner/node/ddl/AlterTablePlanTest.java

```diff
@@ -42,68 +42,60 @@
-            IllegalArgumentException.class,
-            "The relation \"doc.test\" doesn't support changing the setting, because it is included in a logical replication subscription."
-        );
-
-        var oneValidOneInvalid = Settings.builder()
-            .put(SETTING_NUMBER_OF_SHARDS, "1")
-            .put(SETTING_NUMBER_OF_REPLICAS, "1")
-            .build();
+    public void test_alter_allowed_settings_on_a_replicated_table() throws IOException {
+        AlterTablePlan.bind(buildReplicatedTable(SETTING_NUMBER_OF_REPLICAS, "1"), cctx, nctx, Row.EMPTY, SubQueryResults.EMPTY);
+        AlterTablePlan.bind(buildReplicatedTable(INDEX_REFRESH_INTERVAL_SETTING.getKey(), "234ms"), cctx, nctx, Row.EMPTY, SubQueryResults.EMPTY);
+    }
 
+    @Test
+    public void test_alter_forbidden_settings_on_a_replicated_table() throws IOException {
         assertThrowsMatches(
-            () ->   AlterTablePlan.validateTableParametersForReplicatedTables(tableInfo, oneValidOneInvalid),
+            () -> AlterTablePlan.bind(buildReplicatedTable(SETTING_NUMBER_OF_SHARDS, "5"), cctx, nctx, Row.EMPTY, SubQueryResults.EMPTY),
             IllegalArgumentException.class,
-            "The relation \"doc.test\" doesn't support changing the setting, because it is included in a logical replication subscription."
+            "Invalid property \"number_of_shards\" passed to [ALTER | CREATE] TABLE statement"
         );
     }
 
-    private static TableInfo buildTableInfoForSubscribedTable() {
+    private static AnalyzedAlterTable buildReplicatedTable(String setting, String value) {
         var relationName = new RelationName("doc", "test");
         var settings = Settings.builder()
             .put(REPLICATION_SUBSCRIPTION_NAME.getKey(), "sub1")
-            .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), "123ms")
             .put(SETTING_INDEX_VERSION_CREATED.getKey(), Version.CURRENT).build();
 
-        IndexMetadata.Builder indexMetadataBuilder = IndexMetadata
+        var indexMetadataBuilder = IndexMetadata
             .builder(relationName.indexNameOrAlias())
             .settings(settings)
             .numberOfReplicas(0)
             .numberOfShards(1);
 
         var metadata = Metadata.builder().put(indexMetadataBuilder).build();
-        ClusterState state = ClusterState.builder(ClusterName.DEFAULT).metadata(metadata).build();
-        return new DocTableInfoBuilder(createNodeContext(), relationName, state).build();
+        var state = ClusterState.builder(ClusterName.DEFAULT).metadata(metadata).build();
+        var docTableContext = new DocTableInfoBuilder(nctx, relationName, state).build();
+        var genericProperties = new GenericProperties<Symbol>(Map.of(stripIndexPrefix(setting), Literal.of(value)));
+        var alterTable = new AlterTable<>(new Table<>(new QualifiedName("test")), genericProperties);
+        return new AnalyzedAlterTable(docTableContext, alterTable);
```

Comment:
Maybe we could extend (add overload) `SQLExecutor.Builder.addTable()` to take a 2nd argument with settings which we mix in with settings from the create stmt here https://github.com/crate/crate/blob/master/server/src/testFixtures/java/io/crate/testing/SQLExecutor.java#L549.
This way we could utilize the `SQLExecutor.plan` logic to test the alter plan.